### ISSUE TITLE
Cannot setup smtp with latest OSticket

### ISSUE THREAD

**Issue by milindras** [osTicket/osTicket#3715]
I am trying to setup smtp for my new osticket installation but without luck. 
Outgoing server (SMTP)	auth.smtp.1and1.co.uk
Outgoing port (TLS must be activated)	587

I am using php 5.6. 

Is this not supported with TSL?

Many Thanks

### Prerequisites

* [x] Can you reproduce the problem in a fresh installation of the "develop" branch?
* [ ] Do you have any errors in the PHP error log, or javascript console?
* [ ] Did you check the [osTicket forums](http://osticket.com/forum/categories)?
* [ ] Did you [perform a cursory search](https://github.com/osTicket/osTicket/issues) to see if your bug or enhancement is already reported?

For more information on how to write a good [bug report](https://github.com/osTicket/osTicket/wiki/Github-Issues-Guidelines)

### Description

[Description of the bug or feature]

### Steps to Reproduce

1. [First Step]
2. [Second Step]
3. [and so on...]

**Expected behavior:** [What you expected to happen]

**Actual behavior:** [What actually happened]

### Versions

Admin panel -> Dashboard -> Information which also additionally gives you information about you server.

Also, please include the OS and what version of the OS you're running. As well as your browser and browser version.

**Comment by fdellwing** [comment]
There are literally no information provided at all. So, how should anyone help you?

**Comment by milindras** [comment]
Sorry guys I have been searching for a solution for this issue for the past few days and I have seen many topics thoughts you guys know this case. 
my version : osTicket Core, v1.10 and php 5.6

I am trying to setup smtp settings. My hosting is on 1and1 UK.  
When I try that I get below error

Sending Email via SMTP     
Failed to connect to auth.smtp.1and1.co.uk:587 [SMTP: Failed to connect socket: Connection timed out (code: -1, response: )]

I tried many things discussed in few forums such as editing SMTP.php etc but was without luck. 

But today I found this thread - http://osticket.com/forum/discussion/76057/smtp-connection-problems-specific-to-1-1#latest 
 
So I tested below servers and its working fine. Bingo!

Server: mrvnet.kundenserver.de
Port: 25
Authentication: No

Hope this is ok (Authentication = NO). 

Kind Regards
Mili

**Comment by ntozier** [comment]
Please close this issue report then if you have resolved your issue.

**Comment by Kardell** [comment]
I do have exactly the same problem with the latest osTickets and SMTP on 1and1.
I don't think using some free unauthenticated SMTP servers is a solution.
Successful delivery of mail from these is highly unlikely.

**Comment by ntozier** [comment]
Over the last several years of providing support on the forums, I have come to dread when people tell me that they are on 1and1.  Many people have reported issues with their mail servers.  Problems which do not exist on other mail hosts.  So in this case I have to disagree with you.

**Comment by Kardell** [comment]
Hi ntozier,
   At the moment I am trying to configure Amazon WorkMail SMTP server in osTicket that is hosted on 1and1 and it fails.
I have succeeded with IMAP first time.
I provide correct login details and server details according to this document: http://docs.aws.amazon.com/workmail/latest/userguide/using_IMAP_client.html

**Comment by ntozier** [comment]
PHP has failed to connect the socket.  This could be because imap or ssl is not enabled in php.  It could also be a firewall.

**Comment by Kardell** [comment]
but IMAP works fine, do you still think it will be to do with php.ini?
I suspect the worst which is this firewall blocking outgoing SMTP traffic

**Comment by Kardell** [comment]
I have tried a different shared hosting provider and I have exactly the same problem: IMAP works, but not SMTP.
What do I need in PHP for SMTP to work?
Perhaps I will be able to get it working with this provider.

**Comment by ntozier** [comment]
I would recommend that you visit the support forums and post a thread about your situation and your information.  Posting updates on resolved threads will likely go unanswered and you have not provided us with any meaningful environmental details with which to assist you.

Please read before requesting assistance:
http://www.osticket.com/forum/discussion/77765/please-read-before-requesting-assistance#latest

Forums:
http://www.osticket.com/forums

**Comment by Kardell** [comment]
I agree.
The version of PHP has been upgraded to 5.6
This is the list of available modules:
apcu | imagick | mysqli | propro | tidy
apm | imap | mysqlnd | pspell | timezonedb
ares | inotify | ncurses | quickhash | trader
bcmath | interbase | nd_mysql | radius | translit
big_int | intl | nd_mysqli | raphf | uploadprogress
bitset | ioncube_loader | nd_pdo_mysql | rar | uri_template
bz2_filter | ioncube_loader_4 | oauth | recode | uuid
dba | jsmin | oci8 | redis | wddx
dbx | json | odbc | rsync | weakref
dom | ldap | opcache | snmp | xcache_3
doublemetaphone | libevent | pdf | soap | xdebug
eio | libsodium | pdo | sockets | xmlreader
enchant | lzf | pdo_dblib | sourceguardian | xmlrpc
fileinfo | mailparse | pdo_firebird | spl_types | xmlwriter
gd | mbstring | pdo_mysql | ssh2 | xrange
gender | mcrypt | pdo_odbc | stats | xsl
geoip | memcache | pdo_pgsql | stem | yaml
gmagick | memcached | pdo_sqlite | stomp | yaz
gnupg | mongo | pgsql | suhosin | zend_guard_loader
haru | mongodb | phalcon | sybase_ct | zip
htscanner | msgpack | phalcon3 | sysvmsg | zmq
http | mssql | phar | sysvsem |  
igbinary | mysql | posix | sysvshm |  

I have also asked sysadmins to answer my question if they block external SMTP servers or not.

**Comment by Kardell** [comment]
They have confirmed that there is no blocking of external mail servers.

**Comment by Kardell** [comment]
I have registered on your website but have never been able to write a post. I cannot see a button for it in on your forum.

I have moved the system elsewhere where I run everything from a subdomain.
assist.example.com
the email address is on the same subdomain alerts@assist.example.com
I have managed to save all IMAP and SMTP details in the settings with success, but the system is not fetching email, Diagnostics does not work either.
I have tested this email address both ways and send and receive emails.
Not sure why osTicket is able to save mail settings with success, Diagnostics also say that it sent with success, but the system is not fetching emails from IMAP or send from SMTP.
I think the problem is that I use email address on a subdomain. I cannot explain it otherwise.
PHP: 5.6
MySQL: 10

Please advise.
Best regards

**Comment by ntozier** [comment]
I repeat based on the single error you provided it sounds like imap or ssl is not enabled in php, or a firewall. (this includes 1and1's mod_security rules which you would have to have them check their logs and see if it was rejecting the traffic).

Re: the forums: You can email me your email address that you registered with and I can take a look (neil at osticket dot com).  If you put a really lousy reason for wanting access you may have been denied because I or someone else thought that you were a spam bot.

To summarize your previous posts.
Aug 1st - IMAP works SMTP does not.
Sep 1st - IMAP works, SMTP does not.
Today - but the system is not fetching emails from IMAP or send from SMTP.

So which is it?  Are you not fetching and not sending?  or are the Aug 1 and Sept 1 posts correct?

My advice remains: 
- dont use 1and1
- dont post to closed threads
- start a thread on the forums.
- when you do that please provide what you set for your email settings (do not post passwords).

**Comment by Kardell** [comment]
What I can tell you re registration I am not receiving confirmation emails on @live.com account - basically nothing even when I clicked resend, nothing in the Junk folder, but on @gmail.com - all good.
